packages/components/src/components/pin-code/pin-code-utils.spec.ts

```diff
@@ -0,0 +1,231 @@
+import * as pinCodeUtils from './pin-code-utils';
+import * as setAttributeUtils from '../../utils/dom/setAttribute';
+import * as consoleWarnUtils from '../../utils/log/logger';
+import {
+  getStylesWithoutSlottedSelector,
+  initHiddenInput,
+  inputConsistsOfDigits,
+  inputIsSingleDigit,
+  getArrayOfInputValues,
+  syncHiddenInput,
+  warnAboutTransformedInitialValue,
+  getOptimizedValue,
+} from './pin-code-utils';
+import { PinCode } from './pin-code';
+
+describe('getStylesWithoutSlottedSelector()', () => {
+  it('should replace "::slotted()" from Styles objects selectors', () => {
+    const stylesWithSlottedSelector = {
+      '::slotted(input)': {
+        display: 'none',
+      },
+      '@media(hover:hover)': {
+        '::slotted(input:not(:disabled):not(:focus):not([readonly]):hover)': {
+          border: '2px',
+        },
+      },
+    };
+
+    const stylesWithoutSlottedSelector = {
+      input: {
+        display: 'none',
+      },
+      '@media(hover:hover)': {
+        'input:not(:disabled):not(:focus):not([readonly]):hover': {
+          border: '2px',
+        },
+      },
+    };
+
+    expect(getStylesWithoutSlottedSelector(stylesWithSlottedSelector)).toStrictEqual(stylesWithoutSlottedSelector);
+  });
+});
+
+describe('warnAboutTransformedInitialValue()', () => {
+  it('should call consoleWarn() with correct parameters', () => {
+    const warningPrefix = '@Prop() "value" on component <p-pin-code>:';
+    const spy = jest.spyOn(consoleWarnUtils, 'consoleWarn');
+    jest.spyOn(global.console, 'warn').mockImplementation();
```

Comment:
This should do the trick :)
```suggestion
    const spy = jest.spyOn(consoleWarnUtils, 'consoleWarn').mockImplementation();
```